aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -369,13 +369,45 @@ static Tensor nonzero_fallback(const Tensor& self) {
         maskedIndicesTensor = [mpsGraph concatTensors:maskedIndicesTensorArray dimension:0 interleave:YES name:nil];
         coordinatesTensor = [mpsGraph concatTensors:coordinatesTensorArray dimension:0 interleave:YES name:nil];
       }
-
-      MPSGraphTensor* outputTensor = [mpsGraph scatterWithDataTensor:scatterDataTensor
+    
+    MPSGraphTensor* outputTensor;
+
+    // mpsGraph's scatter seems to be, as of this commit, casting int32's (like coordinatesTensor) to float32's and then back,
```

Comment:
This was indeed the case. We are currently doing intermediate indices through F32 conversion which is reducing the range of indexing. We are looking into more efficient way of addressing this issue.

Reply:
Overall the spirit of PR is great.. Thanks for the fix